include/mips-emulator/instruction.hpp

```diff
@@ -272,14 +306,26 @@ namespace mips_emulator {
         }
 
         // FPU T-Type
-        Instruction(const FPUTTypeOp op, const RegisterName rt, const uint8_t fs) {
+        Instruction(const FPUTTypeOp op, const RegisterName rt,
+                    const uint8_t fs) {
             fpu_ttype.cop1 = static_cast<uint8_t>(COPOpcode::e_cop1);
             fpu_ttype.op = static_cast<uint8_t>(op);
             fpu_ttype.rt = static_cast<uint8_t>(rt);
             fpu_ttype.fs = fs & 31;
             fpu_ttype.zero = 0;
         }
 
+        // Special3 R-Type
+        Instruction(const Special3Func func, const Special3RTypeOp op,
+                    const RegisterName rd, const RegisterName rt) {
+            special3_rtype.func = static_cast<uint8_t>(func);
+            special3_rtype.op = static_cast<uint8_t>(op);
+            special3_rtype.rd = static_cast<uint8_t>(rd);
+            special3_rtype.rt = static_cast<uint8_t>(rt);
+            special3_rtype.zero = 0;
```

Comment:
I think there are some instructions that use this field, but at a glance i don't think we need to implement them it should be fine to just set this to 0.